ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerSessionConfiguration.scala

```diff
@@ -7,13 +7,14 @@ import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantSessio
 import com.daml.ledger.api.tls.TlsConfiguration
 
 private[testtool] final case class LedgerSessionConfiguration(
-    participants: Vector[(String, Int)],
+    private val participantAddresses: Vector[(String, Int)],
+    private val ssl: Option[TlsConfiguration],
+    private val partyAllocation: PartyAllocationConfiguration,
     shuffleParticipants: Boolean,
-    ssl: Option[TlsConfiguration],
-    partyAllocation: PartyAllocationConfiguration,
 ) {
-  def forParticipant(hostAndPort: (String, Int)): ParticipantSessionConfiguration = {
-    val (host, port) = hostAndPort
-    ParticipantSessionConfiguration(host, port, ssl, partyAllocation)
-  }
+  lazy val participants: Vector[ParticipantSessionConfiguration] =
+    participantAddresses.map {
+      case (host, port) =>
+        ParticipantSessionConfiguration(host, port, ssl, partyAllocation)
+    }
```

Comment:
```suggestion
    for ((host, port) <- participantAddresses) yield
      ParticipantSessionConfiguration(host, port, ssl, partyAllocation)
```
A `for`-comprehension probably makes the destructuring a bit more terse.

Reply:
If you accept this suggestion, you may want to give a second spin at `scalafmt`.

Reply:
Much nicer, thank you.